tests/thread_test.c

```diff
@@ -28,12 +28,19 @@ static void s_thread_fn(void *arg) {
 
 static int s_test_thread_creation_join_fn(struct aws_allocator *allocator, void *ctx) {
     (void)ctx;
+    aws_common_library_init(allocator);
     struct thread_test_data test_data;
 
     struct aws_thread thread;
     aws_thread_init(&thread, allocator);
 
-    ASSERT_SUCCESS(aws_thread_launch(&thread, s_thread_fn, (void *)&test_data, 0), "thread creation failed");
+    struct aws_thread_options thread_options = *aws_default_thread_options();
+    /* there should be at least 1 cpu on any machine running this test. Just bind that to make sure that code
+     * path is exercised. */
+    thread_options.cpu_id = 0;
```

Comment:
Should we export a function that lets you know how many CPUs there are, and then test this more exhaustively?

Reply:
we will, the next iteration is to add numa node and cpu iteration. Then we'll connect them together.